app/ts-monitor/collector/report.go

```diff
@@ -102,13 +102,14 @@ type ReportJob struct {
 	reportStat *ReportStat
 }
 
-func NewReportJob(addr, database, storeRP string, rpDuration time.Duration, gzipped, compress bool, logger *logger.Logger, errLogHistory string) *ReportJob {
+func NewReportJob(addr, database, storeRP, username, password string, useHttps bool, rpDuration time.Duration, gzipped, compress bool, logger *logger.Logger, errLogHistory string) *ReportJob {
+	protocol := map[bool]string{true: "https", false: "http"}[useHttps]
 	mr := &ReportJob{
 		storeDatabase: database,
 		storeRP:       storeRP,
 		storeDuration: rpDuration,
-		writeUrl:      fmt.Sprintf("http://%s/write?db=%s&rp=%s", addr, database, storeRP),
-		queryUrl:      fmt.Sprintf("http://%s/query", addr),
+		writeUrl:      fmt.Sprintf("%s://%s/write?db=%s&rp=%s&u=%s&p=%s", protocol, addr, database, storeRP, username, password),
+		queryUrl:      fmt.Sprintf("%s://%s/query?u=%s&p=%s", protocol, addr, username, password),
```

Comment:
`queryUrl` should not be set protocol and auth, it may not be the same source as `writeUrl`.